### DISCUSSION TITLE
Provide Support for configurationParameters Arguments

### DISCUSSION THREAD

**Discussion by AndrewQuartic** [Esri/arcgis-powershell-dsc#479] [Ideas]
If we want to incorporate this DSC package into an Azure ARM template `Microsoft.Compute/virtualMachines/extensions`. We would like to pass configurationArguments directly when calling `Invoke-ArcGISConfiguration`.

Can you provide support for both deserializing from json and also accepting a complex object that abides by the configuration schema you've defined? 

Alternatively, could you expose `Invoke-DSCJob` and other related functions to allow us to properly interface with your submodules in the order we define and pass arguments through?

**Comment by AndrewQuartic** [comment]
@spitzerr

**Reply by AndrewQuartic** [reply]
Unless I am misunderstanding this DSC, my chief concern is that I could see scenarios where we would need to do more than `Install`, `InstallLicense`, and `InstallLicenseConfigure` through the `Invoke-ArcGISConfiguration` function. We may seek to InstallLicense each component, but then configure via federation. In that scenario, we do not have a clear understanding of how to implement without lower level access to submodules. We should at least have the option to attempt.

**Reply by nshampur** [reply]
@AndrewQuartic - maybe this is what you are looking to achieve 

https://github.com/Esri/arcgis-azure-templates/blob/main/Releases/11.1/Templates/basedeployment-single-tier.json#L1828-L1859

This is what happens under the hood for ArcGIS Cloud Builder for Microsoft Azure. Above for single tier base deployment.

**Reply by AndrewQuartic** [reply]
Yes, that is our goal. I assume that `variables('dscExtensionArchiveFileName'))` references the same DSC in this repo. If I were to reference a zip DSC of this repository could I feed it the command arguments in a similar fashion as the above template applies? 

Also, I don't like the idea of running an Azure Pipelines build agent then also running DSC on a single machine as a proto-build server. My goal would be to run the same steps I see possible inside of DSCResources but as determined by my specific pipeline use case.

**Reply by AndrewQuartic** [reply]
I am starting to understand this ARM that you sent more. This looks like it could work, and I presume I can reference the script and function for each task I need.

**Reply by AndrewQuartic** [reply]
Also if I run the same configuration DSC but add in the federation piece during a second run, will it skip install steps already applied and go straight to federation or how do you handle that piece?
